### ISSUE TITLE
Duplicate `by-method` function

### ISSUE THREAD

**Issue by danielbraun** [clojure-liberator/liberator#251]
The `by-method` function is defined twice:

1.  https://github.com/clojure-liberator/liberator/blob/1a2a510d60eb5020b028a56619b94fd642cef02d/src/liberator/util.clj#L51
2. https://github.com/clojure-liberator/liberator/blob/b18447eb4c357d93ca46f59dbb67fb14586a5783/src/liberator/core.clj#L614

They behave differently:
* `util/by-method`: Accepts keyword arguments, returns nil if method wasn't defined
* `core/by-method`: Accept a map, return context if method wasn't defined. Due to deep merging of return values, all sequential values in the context are mutated. (Bad behavior, caused me a bug)

I assume the "better" one is `util/by-method`. Can we then at least deprecate the one in core?

**Comment by ordnungswidrig** [comment]
Good catch and sorry for not looking at this bug earlier. Can you describe the problem with core/by-method more detailed? I don't see where deep merging would kick in here. The functions seem similar to me except keyword args and handling of `:any`.

Maybe you could past an example or -- even better -- write a little test?
